sound/soc/sof/loader.c

```diff
@@ -210,30 +211,51 @@ static int load_modules(struct snd_sof_dev *sdev, const struct firmware *fw)
 int snd_sof_load_firmware_memcpy(struct snd_sof_dev *sdev,
 				 const struct firmware *fw)
 {
+	struct snd_sof_pdata *plat_data = dev_get_platdata(sdev->dev);
 	int ret;
 
+	/* set code loading condition to true */
+	sdev->code_loading = 1;
+
+	ret = request_firmware(&plat_data->fw,
+			       plat_data->machine->sof_fw_filename, sdev->dev);
+
+	if (ret < 0) {
+		dev_err(sdev->dev, "error: request firmware failed err: %d\n",
+			ret);
+		return -EINVAL;
```

Comment:
return ret? Why do we need to translate to -EINVAL here?

Reply:
we should return ret directly, let me change.